lib/db-manager.js

```diff
@@ -38,9 +38,9 @@ var dbManager = module.exports = {
       return dbManager.updatePull(pull).then(function() {
          var updates = [];
 
-         if (pull.commitStatus) {
+         if (pull.commitStatuses) {
             updates.push(
-               dbManager.updateCommitStatus(pull.commitStatus)
+               pull.commitStatuses.forEach(status => dbManager.updateCommitStatus(status))
```

Comment:
`forEach()` doesn't return a promise... it doesn't return anything I think. Maybe you mean `.map()` ? and then a splat?

Reply:
Yeah push call should have been moved in the forEach